packages/@magic-ext/aptos/src/MagicAptosWallet.ts

```diff
@@ -0,0 +1,100 @@
+import type {
+  AccountInfo,
+  AdapterPlugin,
+  NetworkInfo,
+  SignMessagePayload,
+  SignMessageResponse,
+} from '@aptos-labs/wallet-adapter-core';
+import { TxnBuilderTypes, Types } from 'aptos';
+import type { Magic } from 'magic-sdk';
+import { AptosExtension } from '.';
+import { APTOS_NETWORKS, APTOS_NODE_URLS, APTOS_WALLET_NAME, ICON_BASE64 } from './constants';
+
+export class MagicAptosWallet implements AdapterPlugin {
+  readonly name = APTOS_WALLET_NAME;
+  readonly url = 'https://magic.link/';
+  readonly icon = ICON_BASE64;
+
+  readonly providerName = 'magicWallet';
+
+  provider: Magic<[AptosExtension]>;
+  private accountInfo: AccountInfo | null;
+
+  constructor(magic: Magic<[AptosExtension]>) {
```

Comment:
i think email should be passed in here, instead of us creating a different "connectWithMagicLink" function

Reply:
sounds good :) 
fixed it. [a35b6b1](https://github.com/magiclabs/magic-js/pull/529/commits/a35b6b13e471b332625932fcda69673c0cabd41e)